### ISSUE TITLE
Crashing with Index was outside the bounds of the array.

### ISSUE THREAD

**Issue by WickedSchnitzel** [anegostudios/VintageStory-Issues#3887]
### Game Version

1.19.8

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

When trying to search/ type in the creative inventory (letter c, maybe others)

### Logs

Running on 64 bit Windows 10.0.19045.0 with 32723 MB RAM
Game Version: v1.19.8 (Stable)
22.05.2024 16:05:12: Critical error occurred
Loaded Mods: aculinaryartillery@1.1.4, ancientarmory@1.0.0, bettercrates@1.7.0, Ceramos@0.5.1, chiseltools@1.11.5, decor@1.0.15, extrachests@1.8.0, fantasycreatures@0.7.1, morecropsfix@1.0.6, fixedpaths@1.0.1, flourbags@1.0.2, foundrymod@1.0.0, giantgouramis@1.0.2, ClaimsRadar@1.1.0, mannequinstand@1.0.3, millwright@1.1.5, moreicons_cs@1.1.0, peacockbasses@1.1.2, primitivesurvival@3.5.5, spyglass@0.5.1, swordz@1.1.7, translocatorengineeringredux@1.5.1, truesunfishes@1.1.2, game@1.19.8, weaponpackalphaunoff@1.5.1, ancienttools@1.5.18, animalcages@3.0.2, beehivekiln@1.5.3, betterfirepit@1.1.4, betterjonasdevices@1.1.0, betterruins@0.3.5, canclaims@1.0.3, canjewelry@0.2.29, carryon@1.7.4, commonlib@2.3.7, composter@1.1.0, expandedfoods@1.6.9, fairplayguardian@1.1.7, fromgoldencombs@1.4.30, furniturelibrary@1.0.11, hangingbaskets@1.1.0, heatretention@1.0.2, herbarium@1.2.0, sailboat@1.3.0, leaderstats@1.0.29, livemap@0.0.4, medievalexpansion@3.13.1, meteoricexpansion@1.2.5, morepiles@1.5.0, moreplaster@1.0.1, pelaguswinds@1.0.0, petai@2.2.4, prospecttogether@1.3.0, rivers@2.4.0, simplestep@1.1.7, stonebakeoven@1.1.3, tentbag@1.1.2, th3dungeon@0.2.1-rc.1, thecritterpack@0.8.9, usefulstuff17@1.3.1, vinconomy@0.2.7, creative@1.19.8, vsinstruments@1.2.7, vsquest@1.2.0, vsserverpanel@0.1.4, survival@1.19.8, webcartographercolorexporter@0.5.2, workbenchexpansion@1.8.0, xlib@0.8.3-pre.1, youvegotmail@1.0.7, awearablelight@1.1.1, cob2@1.6.1, cobserver@1.0.1, em@2.6.0, linearpower@0.2.1, playercorpse@1.9.0, rifteye@0.3.6, stonequarry@3.2.1, vanity@2.1.2, wildcraftfruit@1.2.0, wolftaming@2.1.2, xskills@0.8.5-pre.1, bricklayers@2.5.3, cob@1.8.5, vsvillagecob@0.2.0
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.GameContent.DynFoliageProperties.GetOrLoadTexture(ICoreClientAPI capi, String key) in VSSurvivalMod\Systems\FruitTree\BEFruitTreePart.cs:line 40
   at Vintagestory.GameContent.BlockFruitTreeBranch.get_Item(String textureCode) in VSSurvivalMod\Systems\FruitTree\BlockFruitTreeBranch.cs:line 53
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElement(Int32 indexForLogging, MeshData meshdata, ShapeElement element, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 394
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 306
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 306
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShape(Shape shapeBase, MeshData& modeldata, Vec3f wholeMeshRotation, Vec3f wholeMeshOffset, Single wholeMeshScale, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 302
   at Vintagestory.GameContent.BlockFruitTreeBranch.OnBeforeRender(ICoreClientAPI capi, ItemStack itemstack, EnumItemRenderTarget target, ItemRenderInfo& renderinfo) in VSSurvivalMod\Systems\FruitTree\BlockFruitTreeBranch.cs:line 142
   at Vintagestory.Client.NoObf.InventoryItemRenderer.GetItemStackRenderInfo(ClientMain game, ItemSlot inSlot, EnumItemRenderTarget target, Single dt) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 315
   at Vintagestory.Client.NoObf.InventoryItemRenderer.RenderItemstackToGui(ItemSlot inSlot, Double posX, Double posY, Double posZ, Single size, Int32 color, Single dt, Boolean shading, Boolean origRotate, Boolean showStackSize) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 194
   at Vintagestory.API.Client.GuiElementItemSlotGridBase.RenderInteractiveElements(Single deltaTime) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Inventory\GuiElementItemSlotGridBase.cs:line 509
   at Vintagestory.API.Client.GuiComposer.Render(Single deltaTime) in VintagestoryApi\Client\UI\GuiComposer.cs:line 716
   at Vintagestory.API.Client.GuiDialog.OnRenderGUI(Single deltaTime) in VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 385
   at Vintagestory.Client.NoObf.GuiManager.OnRenderFrameGUI(Single deltaTime) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 318
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 186
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 809
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 1013
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

**Comment by JustARGuy** [comment]
Having same issue on 1.19.7. If it will help, this error is guaranteed if in creative you'll go to "Flora"
Here's my log:
Running on 64 bit Windows 10.0.19045.0 with 16345 MB RAM
Game Version: v1.19.7 (Stable)
23.05.2024 9:11:58: Critical error occurred
Loaded Mods: aculinaryartillery@1.1.3, chiseltools@1.11.1, landformoverhaul@0.0.2, craftingjonas@1.0.6, decor@1.0.15, floralzoneseastasiaticregion@1.0.1, japanesemod@1.2.1, leatherminingbag@1.0.1, mannequinstand@1.0.3, MoreTorchHolders@1.0.0, primitivesurvival@3.5.0, qptech@1.15.1, she1fish@1.2.0, sortablestorage@2.2.3, temporal_gears_stack@1.0.0, game@1.19.7, zoombuttonreborn@1.6.0, ancienttools@1.5.17, animationmanagerlib@0.7.5-dev.1, betterjonasdevices@1.1.0, betterruins@0.3.4, blocksoverlay@3.2.2, helblockpick@2.3.0, carryon@1.7.4, clipfix@2.0.2, commonlib@2.3.7, compostgrass@1.0.0, danatweaks@2.3.0, electricity@0.0.11, expandedfoods@1.6.9, fruittreebranchfix@1.0.0, hudclock@3.4.0, sailboat@1.3.0, knapster@2.9.3, levelup@1.1.8-rc.2, meteoricexpansion@1.2.5, morepiles@1.5.0, norottinghides@1.0.0, oregolems2@0.1.2, prospecttogether@1.3.0, stonebakeoven@1.1.3, storageoptions@1.0.2, th3dungeon@0.2.1, vtpp@1.1.1, creative@1.19.7, vsimgui@1.1.3, survival@1.19.7, xlib@0.8.1, awearablelight@1.1.1, bullseye-continued@2.5.8, tradercamps@1.1.4, configlib@1.1.1, electricityaddon@0.0.2, electricityextensions@0.0.6, em@2.5.2, fsmlib@0.4.0-dev.1, stonequarry@3.2.1, xskills@0.8.1, maltiezcrossbows@0.2.2, maltiezfirearms@0.5.0, tailorsdelight@1.4.3, banners@0.1.0-pre.1
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.GameContent.DynFoliageProperties.GetOrLoadTexture(ICoreClientAPI capi, String key) in VSSurvivalMod\Systems\FruitTree\BEFruitTreePart.cs:line 40
   at Vintagestory.GameContent.BlockFruitTreeBranch.get_Item(String textureCode) in VSSurvivalMod\Systems\FruitTree\BlockFruitTreeBranch.cs:line 53
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElement(Int32 indexForLogging, MeshData meshdata, ShapeElement element, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 394
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 306
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 306
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShape(Shape shapeBase, MeshData& modeldata, Vec3f wholeMeshRotation, Vec3f wholeMeshOffset, Single wholeMeshScale, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 302
   at Vintagestory.GameContent.BlockFruitTreeBranch.OnBeforeRender_Patch1(BlockFruitTreeBranch this, ICoreClientAPI capi, ItemStack itemstack, EnumItemRenderTarget target, ItemRenderInfo& renderinfo)
   at Vintagestory.Client.NoObf.InventoryItemRenderer.GetItemStackRenderInfo(ClientMain game, ItemSlot inSlot, EnumItemRenderTarget target, Single dt) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 315
   at Vintagestory.Client.NoObf.InventoryItemRenderer.RenderItemstackToGui(ItemSlot inSlot, Double posX, Double posY, Double posZ, Single size, Int32 color, Single dt, Boolean shading, Boolean origRotate, Boolean showStackSize) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 194
   at Vintagestory.API.Client.GuiElementItemSlotGridBase.RenderInteractiveElements(Single deltaTime) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Inventory\GuiElementItemSlotGridBase.cs:line 509
   at Vintagestory.API.Client.GuiComposer.Render(Single deltaTime) in VintagestoryApi\Client\UI\GuiComposer.cs:line 716
   at Vintagestory.API.Client.GuiDialog.OnRenderGUI(Single deltaTime) in VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 385
   at Vintagestory.Client.NoObf.GuiManager.OnRenderFrameGUI(Single deltaTime) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 318
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 186
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 809
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 1013
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

**Comment by Craluminum2413** [comment]
@WickedSchnitzel  @JustARGuy Does it happen without any mods installed?

**Comment by JustARGuy** [comment]
So I found something. While exploring I found a pear tree... without an actual tree. It is just a bunch of floating leaves. Hitboxes of the pear tree's base is still can be seen though. Same thing with golden and red apple trees. I guess that one of the mods is breaking the fruit tree's texrures. So that's why when trying to search something in creative inventory the game crashes.

But that's just a theory
